modules/ROOT/pages/salesforce/salesforce-connector-97.adoc

```diff
@@ -0,0 +1,1274 @@
+
+=== Query
+
+* Even though you can see the fields of an sObject and their corresponding types via DataSense, the `Query` operation returns all fields as `String`.
+
+* If you want to use the actual type of the field, you must convert that field to the desired type using a Transform (or Transform Message) component.
+
+* Although `CreatedDate` field appears as `dateTime`, the query returns a `String` representing the date.
+
+* To actually use the field as a `dateTime`, you can configure it using Transform Message.
+
+* To store `Date` and `dateTime` fields, you can use DataWeave expressions to create `Date` and `Calendar` Java objects.
+
+=== Insert values into a Salesforce Drop-down
+
+* Be aware that inserting dependent values into an existing drop-down list field in Salesforce does not always work. Test to confirm functionality.
+
+=== Evaluate Values in a Salesforce Drop-down
+
+* If you are evaluating against a value in an existing drop-down list field in Salesforce, be sure to use the exact value in the dropdown. For example, if you use the value `US` to evaluate against the contents of a drop-down list that contains the value `USA`, the evaluation works, but you end up with two values in the dropdown: one for `US` and one for `USA`.
+
+=== Currency
+
+* Currency values cannot exceed 18 characters in length.
+* When working with multiple currencies, be aware of which currency your sObject uses so that you avoid inaccurate entries. The default currency matches the location at the organization level.
+
+=== Limits on API Calls
+
+* You need to know the rate limiting policy that applies to your account so that you do not exceed the number of allotted API calls per day.
+
+=== Opportunity Object
+
+When extracting data from an `Opportunity`, be aware that a "quarter" is not relative to a calendar year. A "quarter" in this context is relative to the financial year of the organization.
+
+== Example: Accept and Transform Data
+
+image::salesforce/salesforce-outbound.png[sfdc_outbound]
+
+* HTTP connector: Accepts data from HTTP requests.
+* Transform Message: Transforms data structure and format to produce the output Salesforce connector expects.
+* Salesforce connector: (Outbound) Connects with Salesforce and performs an operation to push data to Salesforce.
+
+=== Inbound Scenario
+
+image::salesforce/salesforce-inbound.png[query_inbound]
+
+. Scheduler connector: Triggers flow according to configuration.
+. Salesforce connector: Connects with Salesforce, and returns an InputStream with the query results.
+. Transform Message: Transforms data structure and format to produce output the File endpoint expects.
+. File connector: Records data in a file, such as a CSV and saves it to a user-defined directory or location.

```

Comment:
Need a comma before "CSV." I would say "Records data in a file, such as a CSV file, ..."